src/main/java/Deadline.java

```diff
@@ -0,0 +1,13 @@
+public class Deadline extends Task {
+    protected String dateAndTime;
+
+    public Deadline(String description, String dateAndTime) {
+        super(description);
+        this.dateAndTime = dateAndTime;
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + this.getStatusIcon() + " " + super.description + "(by:" + this.dateAndTime + ")";
```

Comment:
Following the DRY Principle, instead of using `this.getStatusIcon() + " " + super.description`, I feel that it will be better to use the toString method in the parent class `Task`, since the code is repeated for all 3 types of tasks.
```suggestion
        return "[D]" + super.toString() + "(by:" + this.dateAndTime + ")";
```